### ISSUE TITLE
[Magellan]  On page load (without any anchor) first menu item is always rendered active, even if anchor is not yet active. 

### ISSUE THREAD

**Issue by phifa** [zurb/foundation-sites#7308]
**How can we reproduce this bug?**
1. Go to the sample page: http://stage.dasachtegebot.de/magellan/lorem.html 
2. Click on the menu items in the Top Bar. 
3. At the destination, the headlines are covered with the Top Bar. 

**What did you expect to happen?**
I expect the scrolling to end where the section starts, including the Fixed Top Bar! So the height of the Magellan Container needs to be included in the scroll position, or there needs to be a scroll offset option. When I add **threshold: 290**, the scrolling is correct (for one media query), but the magellan active stops working properly. I have to use **threshold: -1** though for it to work at all... But I guess that is another issue. Here is my app.js: http://stage.dasachtegebot.de/magellan/assets/js/app.js

**What happened instead?**
Magellan scrolls too far. 

**Test case**
http://stage.dasachtegebot.de/magellan/lorem.html

**Comment by zurbchris** [comment]
I see the problem. If you come up with a solution, I'd love to see a PR. Otherwise this fix will take a bit to get to. Thanks for pointing it out.

**Comment by phifa** [comment]
Too bad, I have no idea how to do this. But, I can report another error :smile: 
As you can see on the example page, once the first nav item is marked "active" and you scroll back to the top, it always stays active. 
http://stage.dasachtegebot.de/magellan/lorem.html

**Comment by TvrtkoM** [comment]
Hi all,
@phifa it works for me. The links are changing 'active' class when i scroll up. What browser are you using?

Regarding a feature maybe add an offset option which acts on on *Foundation.MediaQuery.current* state. For example options like offset_small, offset_medium ... and if its just *offset* it's value will be used for all screens.

Another solution comes to mind is referencing another element and calculate offset on event time, but this seems weird.

**Comment by phifa** [comment]
Well, they render active but the first item stays active, even if I scroll way before it. See the screenshot.
![blue](https://cloud.githubusercontent.com/assets/3061129/11554297/177de5cc-9997-11e5-9219-80ee9d447263.JPG)

**Comment by zurbchris** [comment]
That's the default, to add the active class to the first list item. I'll put it on the list of minor bug fixes unless you want to submit a pr to solve it :smile:

**Comment by phifa** [comment]
Not sure if I can do that, but ok. It definitely needs to be done. I recently used BS for a website and there it works out of the box: http://www.durr.com/challenge/

**Comment by creativereason** [comment]
I'm encountering this same issue, would switching to the RC resolve it? Is that version stable?

**Comment by jtborger** [comment]
I seem to have the same problem. I have a sticky topbar with magellan below in a submenu.
the position of the target is not right. It seems that barOffset is not taken into account.
I've made a patch for myself but its quite dirty as I am not too familiar with bower for compilation of the different js files into the main file.

Just for the record and for now:
in the foundation.magellan.js line 42 it wants to read out the scrollPos by getting the pageYOffset.
Every line (two...) in the magellan class with pageYOffset I replaced with a function getYOffset()

this function just does add the barOffset.
  ```
function getYOffset() {
	  var offset = parseInt(window.pageYOffset,10);
	  if(undefined != this.options.barOffset)
	  {
		  offset+=this.options.barOffset;
	  }
	  return offset;
  }
```

So line 42 becomes:
`   this.scrollPos = this.getYOffset();`
and line 135:
`var winPos = /*scrollPos ||*/ this.getYOffset()//parseInt(window.pageYOffset, 10), 
        curIdx;`
Sure I miss some other stuff but for now just hope it gives someone a hint...

**Comment by phifa** [comment]
Would really appreciate if this gets fixed. Basically the first item should only render active, when it is in view, just like the other items.

**Comment by andycochran** [comment]
This is not specifically an issue with Magellan. It's an issue with any page anchor and a sticky element: http://codepen.io/andycochran/pen/qZMJpE?editors=1100

**Comment by andycochran** [comment]
I updated that [CodePen](http://codepen.io/andycochran/pen/qZMJpE?editors=1100) to show a quick fix. JavaScript could prepend a `span` to all anchors that are linked to from Menus.

**Comment by phifa** [comment]
I changed the title of this issue to reflect more the nature of the problem. @andycochran does that makes sense?

**Comment by andycochran** [comment]
Ack! Sorry. I misunderstood the issue and derailed the conversation. Thanks for clarifying the issue with the title change.

**Comment by danmitchell-** [comment]
+1

**Comment by rawiki** [comment]
Hi, to stop the first megellan menu item being rendered active before it's reached an anchor point you can modify the function _updateActive(). The variable curIdx holds the array position of the item the active class will be added to and if winPos < this.points[0] is true it means that an anchor has not been reached so by changing curIdx from 0 to undefined in the if else statement it will fix the issue. This will also let you scroll back up over the first anchor point and there will be no active classes. Modifying this variable doesn't look like it will affect the rest of the code at all but I haven't run any proper tests. I have also submitted a pull request.

**Comment by kball** [comment]
Fixed by #9243.  Thanks @rawiki!

**Comment by rawiki** [comment]
@kball No problem, glad that's fixed it
